write-fonts/src/font_builder.rs

```diff
@@ -93,35 +127,99 @@ impl<'a> FontBuilder<'a> {
         self.tables.contains_key(&tag)
     }
 
+    /// Returns the builder's table tags in the order recommended by the OpenType spec.
+    ///
+    /// Table tags not in the recommended order are sorted lexicographically, and 'DSIG'
+    /// is always sorted last.
+    /// The presence of the 'CFF ' table determines which of the two recommended orders is used.
+    /// This matches fontTools' `sortedTagList` function.
+    ///
+    /// See:
+    /// <https://learn.microsoft.com/en-us/typography/opentype/spec/recom#optimized-table-ordering>
+    /// <https://github.com/fonttools/fonttools/blob/8d6b2f8f87637fcad8dae498d32eae738cd951bf/Lib/fontTools/ttLib/ttFont.py#L1096-L1117>
+    pub fn ordered_tags(&self) -> Vec<Tag> {
+        let recommended_order: &[Tag] = if self.contains(Tag::new(b"CFF ")) {
+            &RECOMMENDED_TABLE_ORDER_CFF
+        } else {
+            &RECOMMENDED_TABLE_ORDER_TTF
+        };
+        // Sort tags into three groups:
+        //   Group 0: tags that are in the recommended order, sorted accordingly.
+        //   Group 1: tags not in the recommended order, sorted alphabetically.
+        //   Group 2: 'DSIG' is always sorted last, matching fontTools' behavior.
+        let mut ordered_tags: Vec<Tag> = self.tables.keys().copied().collect();
+        let dsig = Tag::new(b"DSIG");
+        ordered_tags.sort_unstable_by_key(|rtag| {
+            let tag = *rtag;
+            if tag == dsig {
+                (2, 0, tag)
+            } else if let Some(idx) = recommended_order.iter().position(|t| t == rtag) {
```

Comment:
my "accidentally quadratic" sense is tingling, although we know that the values of M & N are bounded and small so it's fine...